### ISSUE TITLE
Exception during Navigation on Android

### ISSUE THREAD

**Issue by blaketarter** [NativeScript/NativeScript#1313]
I've found an issue on the latest release (`1.5.2`) in navigating between pages on Android, where the code works just fine on iOS but fails on any Android platform I try it on.

A snippet of the code I'm using to navigate is 
```
  var topmost = frameModule.topmost();

  var navigationEntry = {
    moduleName: 'views/comment/comment',
    context: {
      post: postData
    },
    animated: true
  };

  topmost.navigate(navigationEntry);
```

and on Android the exception that gets thrown is this

```
com.tns.NativeScriptException: 
Calling js method onItemClick failed

Error: 
Error calling module function 

Error: com.tns.NativeScriptException: No value for main
    com.tns.Module.resolvePathHelper(Module.java:164)
    com.tns.Module.resolvePath(Module.java:62)
    com.tns.Platform.callJSMethodNative(Native Method)
    com.tns.Platform.dispatchCallJSMethodNative(Platform.java:799)
    com.tns.Platform.callJSMethod(Platform.java:700)
    com.tns.Platform.callJSMethod(Platform.java:680)
    com.tns.Platform.callJSMethod(Platform.java:670)
    com.tns.gen.android.widget.AdapterView_OnItemClickListener_ftns_modules_ui_list-view_list-view_l63_c45__.onItemClick(android.widget.AdapterView$OnItemClickListener.java)
    android.widget.AdapterView.performItemClick(AdapterView.java:310)
    android.widget.AbsListView.performItemClick(AbsListView.java:1145)
    android.widget.AbsListView$PerformClick.run(AbsListView.java:3042)
    android.widget.AbsListView$3.run(AbsListView.java:3879)
    android.os.Handler.handleCallback(Handler.java:739)
    android.os.Handler.dispatchMessage(Handler.java:95)
    android.os.Looper.loop(Looper.java:148)
    android.app.ActivityThread.main(ActivityThread.java:5417)
    java.lang.reflect.Method.invoke(Native Method)
    com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
    com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
File: "<unknown>, line: 1, column: 119

StackTrace: 
	Frame: function:'require', file:'', line: 1, column: 120
	Frame: function:'', file:'/data/data/com.blaketarter.tnsreddit/files/app/views/comment/comment.js', line: 9, column: 17
	Frame: function:'require', file:'', line: 1, column: 120
	Frame: function:'resolvePageFromEntry', file:'/data/data/com.blaketarter.tnsreddit/files/app/tns_modules/ui/frame/frame-common.js', line: 61, column: 29
	Frame: function:'Frame.navigate', file:'/data/data/com.blaketarter.tnsreddit/files/app/tns_modules/ui/frame/frame-common.js', line: 126, column: 20
	Frame: function:'listViewItemTap', file:'/data/data/com.blaketarter.tnsreddit/files/app/views/front/front.js', line: 126, column: 11
	Frame: function:'Observable.notify', file:'/data/data/com.blaketarter.tnsreddit/files/app/tns_modules/data/observable/observable.js', line: 118, column: 23
	Frame: function:'ListView._createUI.android.setOnItemClickListener.android.widget.AdapterView.OnItemClickListener.onItemClick', file:'/data/data/com.blaketarter.tnsreddit/files/app/tns_modules/ui/list-view/list-view.js', line: 67, column: 27


File: "<unknown>, line: 1, column: 119

StackTrace: 
	Frame: function:'require', file:'', line: 1, column: 120
	Frame: function:'resolvePageFromEntry', file:'/data/data/com.blaketarter.tnsreddit/files/app/tns_modules/ui/frame/frame-common.js', line: 61, column: 29
	Frame: function:'Frame.navigate', file:'/data/data/com.blaketarter.tnsreddit/files/app/tns_modules/ui/frame/frame-common.js', line: 126, column: 20
	Frame: function:'listViewItemTap', file:'/data/data/com.blaketarter.tnsreddit/files/app/views/front/front.js', line: 126, column: 11
	Frame: function:'Observable.notify', file:'/data/data/com.blaketarter.tnsreddit/files/app/tns_modules/data/observable/observable.js', line: 118, column: 23
	Frame: function:'ListView._createUI.android.setOnItemClickListener.android.widget.AdapterView.OnItemClickListener.onItemClick', file:'/data/data/com.blaketarter.tnsreddit/files/app/tns_modules/ui/list-view/list-view.js', line: 67, column: 27


	at com.tns.Platform.callJSMethodNative(Native Method)
	at com.tns.Platform.dispatchCallJSMethodNative(Platform.java:799)
	at com.tns.Platform.callJSMethod(Platform.java:700)
	at com.tns.Platform.callJSMethod(Platform.java:680)
	at com.tns.Platform.callJSMethod(Platform.java:670)
	at com.tns.gen.android.widget.AdapterView_OnItemClickListener_ftns_modules_ui_list-view_list-view_l63_c45__.onItemClick(android.widget.AdapterView$OnItemClickListener.java)
	at android.widget.AdapterView.performItemClick(AdapterView.java:310)
	at android.widget.AbsListView.performItemClick(AbsListView.java:1145)
	at android.widget.AbsListView$PerformClick.run(AbsListView.java:3042)
	at android.widget.AbsListView$3.run(AbsListView.java:3879)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5417)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

I've tried on multiple versions of Android emulators and devices.

**Comment by NathanaelA** [comment]
I am using the latest Android runtimes (v1.52) & common core (v1.51) with my LiveEdit plugin which uses the same navigate with a options object. Right now, I'm in a lot of auto-restarts of the app; but it appears everything is navigating fine on my Android emulator

That error looks like it can't find the .js file.  Are you sure that "app/views/comment/comment.js" exists in your android project?     
Also another possibility is that if the comment.js file is messed up; when it is required it can fail on the Java side and return nothing which will then fail on the JS side.

**Comment by blaketarter** [comment]
Hmm, the file definitely exists at `platforms/android/src/main/assets/app/views/comment/comment.js`, I'm taking a look at the contents to see what would cause it to fail being required, it passes my jshint though.

**Comment by NathanaelA** [comment]
Hmm, all my requires are using "./somepath/file" with the "./" prefixed.  It might be an issue of absolute vs relative pathing...

**Comment by blaketarter** [comment]
The path to the module should be relative from `app` correct? Even when I change it to `./views/comment/comment` I still get the same error.

However I did just intentionally put an incorrect path in and it throws a different exception. I'll copy that in here real quick.

**Comment by blaketarter** [comment]
```
com.tns.NativeScriptException: 
Calling js method onItemClick failed

Error: Failed to load Page from entry.moduleName: ../../views/comment/comment
File: "/data/data/com.blaketarter.tnsreddit/files/app/tns_modules/ui/frame/frame-common.js, line: 71, column: 12

StackTrace: 
	Frame: function:'resolvePageFromEntry', file:'/data/data/com.blaketarter.tnsreddit/files/app/tns_modules/ui/frame/frame-common.js', line: 71, column: 19
	Frame: function:'Frame.navigate', file:'/data/data/com.blaketarter.tnsreddit/files/app/tns_modules/ui/frame/frame-common.js', line: 126, column: 20
	Frame: function:'listViewItemTap', file:'/data/data/com.blaketarter.tnsreddit/files/app/views/front/front.js', line: 126, column: 11
	Frame: function:'Observable.notify', file:'/data/data/com.blaketarter.tnsreddit/files/app/tns_modules/data/observable/observable.js', line: 118, column: 23
	Frame: function:'android.setOnItemClickListener.android.widget.AdapterView.OnItemClickListener.onItemClick', file:'/data/data/com.blaketarter.tnsreddit/files/app/tns_modules/ui/list-view/list-view.js', line: 67, column: 27


	at com.tns.Platform.callJSMethodNative(Native Method)
	at com.tns.Platform.dispatchCallJSMethodNative(Platform.java:799)
	at com.tns.Platform.callJSMethod(Platform.java:700)
	at com.tns.Platform.callJSMethod(Platform.java:680)
	at com.tns.Platform.callJSMethod(Platform.java:670)
	at com.tns.gen.android.widget.AdapterView_OnItemClickListener_ftns_modules_ui_list-view_list-view_l63_c45__.onItemClick(android.widget.AdapterView$OnItemClickListener.java)
	at android.widget.AdapterView.performItemClick(AdapterView.java:310)
	at android.widget.AbsListView.performItemClick(AbsListView.java:1145)
	at android.widget.AbsListView$PerformClick.run(AbsListView.java:3042)
	at android.widget.AbsListView$3.run(AbsListView.java:3879)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5417)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

**Comment by NathanaelA** [comment]
Hmm, that would say to me that the JS file has something that is causing it to fail. You can try to verify really quick to rename your comment.js file to comment2.js and create a simple comment.js file with something like "console.log('hi');" and see if that works.

**Comment by blaketarter** [comment]
Yeah that's absolutely it, just tried that. So now I just need to diagnose whats causing it to fail on Android and not iOS, while still passing my linter.

**Comment by NathanaelA** [comment]
I submitted a bug a week or so ago where a high ascii character in a comment was causing this exact issue on iOS.    The file ran perfectly on Android; just failed when ran on iOS on a require.  

Use something that can show you all the characters (I use the open source Notepad++ to do this), or a hex editor.   This way you can see if their are any characters in the stream that aren't visible but are causing the issues.

**Comment by blaketarter** [comment]
So I've narrowed it down to an npm module I was using to strip html tags out of strings (https://github.com/ericnorris/striptags), curious that it works elsewhere. Looks like I'll either find a replacement module or look for the issue in his code.

**Comment by blaketarter** [comment]
I don't know if this issue relates to this project, but I found that the npm module was causing an exception in Android because its package.json was missing `"main": "index.js",`, yet it still worked in iOS. Adding it in manually fixes all the issues, I'm going to submit a pull request on that repo.

**Comment by NathanaelA** [comment]
Ah, yeah, I've saw a bug report the v1.5x runtime broke the missing package main functionality again.  It should be fixed in v1.6 - https://github.com/NativeScript/android-runtime/issues/316

**Comment by blaketarter** [comment]
Perfect, thanks
